src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2365,6 +2366,9 @@ private bool InstructionSelfDestruct<TTracing>(EvmState vmState, ref EvmStack<TT
             _state.AddToBalance(inheritor, result, spec);
         }
 
+        if (spec.SelfdestructOnlyOnSameTransaction && !createInSameTx && inheritor.Equals(executingAccount))
+            return true; // dont burn eth when contract is not destroyed per EIP clarification
+
         _state.SubtractFromBalance(executingAccount, result, spec);
         return true;
```

Comment:
Wouldn't:

```
        if (!spec.SelfdestructOnlyOnSameTransaction || createInSameTx || !inheritor.Equals(executingAccount))
        {
            _state.SubtractFromBalance(executingAccount, result, spec);
        }

        return true;
```

be more readable?

Reply:
I like to specify only the condition where the burn does not happen. I find it less confusing than specifying all the conditions that it does burn.